pkg/recorder/recorder.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package recorder
+
+import (
+	"github.com/golang/glog"
+	"github.com/kubernetes-sigs/controller-runtime/pkg/client/config"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	typedcorev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/record"
+)
+
+type RecorderGetFuncType func(string, *runtime.Scheme) record.EventRecorder
+
+var eventBroadcaster record.EventBroadcaster
+
+func GetEventRecorderFor(name string, scheme *runtime.Scheme) record.EventRecorder {
+	if eventBroadcaster == nil {
+		cs := config.GetKubernetesClientSetOrDie()
```

Comment:
We should get the clientset from the manager.Manager, or have users set it on a struct.  All components should use the same client.